I imported a project from door43 that had a new...

Notes
======
I imported a project from door43 that had a new source text that had not yet been downloaded to my device. The import succeeded and I was able to open the project. The crash occurred when I clicked the icon to select sources.

Environment
======
Environment Key | Value
:----: | :----:
version | 11.0
build | 164
UDID | b65ef2f052953fdb
Android Release | 5.0.1
Android SDK | 21
Brand | Lenovo
Device | A10-70F
Stack trace
======
```java
java.lang.NullPointerException: Attempt to read from field 'java.lang.String org.unfoldingword.door43client.models.Translation.resourceContainerSlug' on a null object reference
	at com.door43.translationstudio.ui.translate.ChooseSourceTranslationDialog$9.start(ChooseSourceTranslationDialog.java:256)
	at org.unfoldingword.tools.taskmanager.ManagedTask.run(ManagedTask.java:78)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)

```
Log history
======
```java
3/22/17 9:24 AM  E/com.door43.translationstudio.tasks.PullTargetTranslationTask: Could not get advertised Ref for branch master
org.eclipse.jgit.api.errors.JGitInternalException: Could not get advertised Ref for branch master
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:288)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.pull(PullTargetTranslationTask.java:149)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.start(PullTargetTranslationTask.java:84)
	at com.door43.util.tasks.ManagedTask.run(ManagedTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)

3/22/17 9:24 AM  E/com.door43.translationstudio.tasks.PullTargetTranslationTask: Invalid remote: origin
org.eclipse.jgit.api.errors.InvalidRemoteException: Invalid remote: origin
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:136)
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:261)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.pull(PullTargetTranslationTask.java:149)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.start(PullTargetTranslationTask.java:84)
	at com.door43.util.tasks.ManagedTask.run(ManagedTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: org.eclipse.jgit.errors.NoRemoteRepositoryException: git@git.door43.org:qa98/en-x-demo1_php_text_reg.git: Gogs: Repository does not exist or you do not have access
Repository does not exist: qa98/en-x-demo1_php_text_reg

	at org.eclipse.jgit.transport.TransportGitSsh.cleanNotFound(TransportGitSsh.java:208)
	at org.eclipse.jgit.transport.TransportGitSsh$SshFetchConnection.<init>(TransportGitSsh.java:288)
	at org.eclipse.jgit.transport.TransportGitSsh.openFetch(TransportGitSsh.java:161)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:136)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:122)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1138)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:130)
	... 7 more

3/22/17 9:24 AM  E/com.door43.translationstudio.tasks.PullTargetTranslationTask: git@git.door43.org:qa98/en-x-demo1_php_text_reg.git: Auth fail
org.eclipse.jgit.api.errors.TransportException: git@git.door43.org:qa98/en-x-demo1_php_text_reg.git: Auth fail
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:139)
	at org.eclipse.jgit.api.PullCommand.call(PullCommand.java:261)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.pull(PullTargetTranslationTask.java:149)
	at com.door43.translationstudio.tasks.PullTargetTranslationTask.start(PullTargetTranslationTask.java:84)
	at com.door43.util.tasks.ManagedTask.run(ManagedTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: org.eclipse.jgit.errors.TransportException: git@git.door43.org:qa98/en-x-demo1_php_text_reg.git: Auth fail
	at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:159)
	at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:136)
	at org.eclipse.jgit.transport.TransportGitSsh$SshFetchConnection.<init>(TransportGitSsh.java:262)
	at org.eclipse.jgit.transport.TransportGitSsh.openFetch(TransportGitSsh.java:161)
	at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:136)
	at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:122)
	at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1138)
	at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:130)
	... 7 more
Caused by: com.jcraft.jsch.JSchException: Auth fail
	at com.jcraft.jsch.Session.connect(Session.java:512)
	at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:116)
	... 14 more

     

```

--- a/app/src/main/java/com/door43/translationstudio/ui/translate/ChooseSourceTranslationDialog.java
+++ b/app/src/main/java/com/door43/translationstudio/ui/translate/ChooseSourceTranslationDialog.java
@@ -253,8 +253,10 @@ public class ChooseSourceTranslationDialog extends DialogFragment implements Man
                     String[] sourceTranslationSlugs = App.getOpenSourceTranslations(mTargetTranslation.getId());
                     for (String slug : sourceTranslationSlugs) {
                         Translation st = mLibrary.index.getTranslation(slug);
-                        this.publishProgress(-1,st.resourceContainerSlug);
-                        if (st != null) addSourceTranslation(st, true);
+                        if (st != null) {
+                            this.publishProgress(-1,st.resourceContainerSlug);
+                            addSourceTranslation(st, true);
+                        }
                     }
 
                     List<Translation> availableTranslations = mLibrary.index.findTranslations(null, mTargetTranslation.getProjectId(), null, "book", null, App.MIN_CHECKING_LEVEL, -1);

